File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py


def solution(k, tangerine):
    answer = 0
    counts = []
    dic = dict()
    for i in tangerine:
        if i in dic:
            dic[i] += 1
        else: 
            dic[i] = 1
    dic = dict(sorted(dic.items(), key=lambda x:x[1], reverse=True))
    for i in dic:
        if k <= 0:
            return answer
        k -= dic[i]
        answer += 1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const dic = {};
    tangerine.forEach((el) => dic[el]  = (dic[el] || 0) + 1)
    const arr = Object.values(dic).sort((a, b)=> b-a);
    for (const a of arr) {
        answer++;
        if (k > a) k -= a;
        else break;
    }
    return answer;
}